### PR TITLE
👌️ IMPROVE:  Review of test infrastructure

### PR DESCRIPTION
**Pull request by AakashGfude**
Following the philosophy of these discussions and PRs: https://github.com/executablebooks/jupyter-book/issues/896 
https://github.com/executablebooks/jupyter-book/pull/624

and test suites of https://github.com/executablebooks/MyST-NB

- [x]  create a sphinx testing infra

### CODE REVIEW

**Comment by mmcky** [comment]
thanks @AakashGfude for reviewing the test suites. I only have one comment. Can we have `test_` (rather than `test-`) to be consistent with `_` in the names. It's minor but a little bug bear of mine mixing the two `:-)`

**Comment by AakashGfude** [comment]
@mmcky yes, its a bit out of place, but `pytest sphinx testing` expects patterns like `test-*` for test folders. https://github.com/sphinx-doc/sphinx/issues/7008 .  If I didnt prepend it with `tests-*` it was not able to find it easily. And as can be seen implemented in this testing infra https://github.com/sphinx-doc/sphinxcontrib-jsmath/tree/master/tests/roots .
Probably we can do either of these two things: 
1) change the `_` in the name to `-` as well for consistency
2) start digging more to see how to support the different name.

**Comment by mmcky** [comment]
@AakashGfude then how to [these work](https://github.com/executablebooks/MyST-NB/tree/master/tests/test_execute)

**Comment by AakashGfude** [comment]
hmmm, @mmcky , 'test-*' was used by `testroot` to set dirs
image

MyST-NB does not use this fixture, and instead is somewhat creating its own sphinx env in a sandbox https://github.com/executablebooks/MyST-NB/blob/master/tests/conftest.py , which looks super complex. probably because it wants more control for intricate testing? Not sure. Nevertheless, it seemed really complex for this small app. So, went with a generic one. 
But, I am not sure if there is a simple workaround for this.

**Comment by mmcky** [comment]
Oh I see so your not using `myst_nb` - I was confused from the initial comment in the PR. 

I modified the `EBP` setup for [sphinxcontrib-rst2myst](https://github.com/mmcky/sphinxcontrib-rst2myst/tree/master/tests) -- perhaps take a quick look there. It is pretty simple and fixtures are really nice as you get to see outputs and xml etc.

**Comment by AakashGfude** [comment]
@mmcky gotcha thanks. Was about to comment, that I will ditch using `testroot`, and create the path somewhere else. That test suite is really helpful.

**Comment by mmcky** [comment]
@AakashGfude please tag @najuzilu when you're ready for final review on this.

**Comment by AakashGfude** [comment]
@mmcky sure.

**Comment by AakashGfude** [comment]
@mmcky at present in this directive `algorithm` and `proof` directives has been tested as was originally done by @najuzilu . Since, all the other directives except `proof` have the same directive definition as `algorithm`, I guess basic test for `algorithm` is sufficient to test others as well. I guess we should wait till `coverage` has been set up, which will give us a better idea of what else needs testing.

**Comment by mmcky** [comment]
thanks @AakashGfude can we go ahead and setup `coverage`? Doesn't seem to be included in https://github.com/najuzilu/sphinxcontrib-prettyproof/pull/25.

**Review by mmcky — COMMENTED** [review]
hey @AakashGfude I don't see the updates to `docs` re: test setup. We should add that also to this PR.

**Comment by AakashGfude** [comment]
@mmcky oops, should have started a review after pushing that. editing that right now. 
@mmcky should we setup coverage in this PR or a new one and merge that first?

**Comment by AakashGfude** [comment]
@mmcky have added a basic doc for `testing`. Let me know what to add/delete.

**Comment by AakashGfude** [comment]
sorry my bad, `pytest-cov` is already added to the repo. Running `pytest --cov=sphinxcontrib` gave `97%` coverage, which is pretty good.

**Comment by mmcky** [comment]
is this pytest part of `CI`? -- 97% is excellent

**Comment by mmcky** [comment]
> @mmcky have added a basic doc for `testing`. Let me know what to add/delete.

looks good. Thanks @AakashGfude

**Comment by AakashGfude** [comment]
> is this pytest part of `CI`? -- 97% is excellent

Not part of the CI. But I was assuming we need to link it to a codecov account  to generate reports online. Can we generate xml report locally and make it part of CI without needing to link for now?

**Comment by AakashGfude** [comment]
but anyways we will need the workflow file in https://github.com/najuzilu/sphinxcontrib-prettyproof/pull/25 for adding coverage I reckon.

**Comment by mmcky** [comment]
This should be done via `codecov` -- perhaps there is an org account for `executablebooks`. We can add this after merge.

**Comment by AakashGfude** [comment]
@mmcky , yes I agree, we should do that. Perhaps, for now, we should add test-with-cov https://github.com/executablebooks/jupyter-book/blob/master/.github/workflows/tests.yml#L61 in the workflow https://github.com/najuzilu/sphinxcontrib-prettyproof/pull/25

File: tests/test_algorithm.py [hunk 1]
```diff
@@ -1,70 +1,38 @@
-from pathlib import Path
-from subprocess import run
 from bs4 import BeautifulSoup
+import pytest
 import os
```
Comment (najuzilu) [inline tests/test_algorithm.py:3]: minor edit suggestion: 'os' imported but unused

File: tests/test_proof.py [hunk 2]
```diff
@@ -1,55 +1,27 @@
-from pathlib import Path
-from subprocess import run
 from bs4 import BeautifulSoup
 import pytest
 import os
```
Comment (najuzilu) [inline tests/test_proof.py:3]: minor edit suggestion: 'os' imported but unused

**Review by najuzilu — APPROVED** [review]
Great work @AakashGfude! I'm working on implementing codecov in workflows.

File: tests/test_proof.py (hunk 2 above)
Comment (AakashGfude) [reply tests/test_proof.py:3]: Good catch @najuzilu will remove this and the above one.

**Comment by AakashGfude** [comment]
haha, yeah emojis don't work in the title.

**Review by mmcky — APPROVED** [review]
thanks @AakashGfude this is looking good. @najuzilu please merge when you're ready.

**Pull request merged**